android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -18,16 +19,14 @@ import java.util.UUID
  *
  * @param resources Your application's [Resources] instance.
  * @param loadUrlUseCase A use case that loads a suggestion's URL when clicked.
- * @param includeSponsoredSuggestions Whether to return suggestions for sponsored content.
- * @param includeNonSponsoredSuggestions Whether to return suggestions for web content.
+ * @param providers The [SuggestionProvider]s to query for search suggestions.
  * @param suggestionsHeader An optional header title for grouping the returned suggestions.
  * @param contextId The contextual services user identifier, used for telemetry.
  */
 class FxSuggestSuggestionProvider(
     private val resources: Resources,
     private val loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
-    private val includeSponsoredSuggestions: Boolean,
-    private val includeNonSponsoredSuggestions: Boolean,
+    @get:VisibleForTesting val providers: List<SuggestionProvider>,
```

Comment:
Honestly, I don't see a real reason to not just let this be public since the type is an immutable type and the property is immutable as well.

Reply:
Thanks—I reverted this change as part of adding the component-specific FML, but I'll keep it in mind for the future!